sycl/source/detail/device_image_impl.hpp

```diff
@@ -0,0 +1,127 @@
+//==------- device_image_impl.hpp - SYCL device_image_impl -----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl/context.hpp>
+#include <CL/sycl/detail/common.hpp>
+#include <CL/sycl/detail/pi.h>
+#include <CL/sycl/device.hpp>
+#include <CL/sycl/kernel_bundle.hpp>
+#include <detail/device_impl.hpp>
+#include <detail/kernel_id_impl.hpp>
+#include <detail/program_manager/program_manager.hpp>
+
+#include <algorithm>
+#include <cassert>
+#include <cstring>
+#include <memory>
+#include <vector>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace detail {
+
+class device_image_impl {
```

Comment:
Added comments.
Usually for impl counterpart with just add "_impl" at the end of the class name spec suggests, so renaming it would be a confusion as well.

Reply:
OK, I see. `device_image` is the name from the spec, hence `device_image_impl`. That's OK then.
I still think `device_image` is a bad name, but that's a spec issue.